src/types.rs

```diff
@@ -1,5 +1,65 @@
 use std::collections::HashMap;
 
+use std::fmt;
+use std::cmp;
+use serde::*;
+use serde::de::{Visitor, Error, Unexpected};
+
+#[derive(Clone, Debug)]
+pub struct Timestamp {
+    string_repr: String,
+    float_repr: f64,
+}
+
+impl cmp::PartialEq for Timestamp {
+    fn eq(&self, other: &Timestamp) -> bool {
+        self.float_repr.eq(&other.float_repr)
+    }
+}
+
+impl cmp::PartialOrd for Timestamp {
+    fn partial_cmp(&self, other: &Timestamp) -> Option<cmp::Ordering> {
+        self.float_repr.partial_cmp(&other.float_repr)
+    }
+}
+
+/// Deserialize a maybe-string timestamp into a Timestamp.
+pub fn deserialize_timestamp<'d, D: Deserializer<'d>>(d: D) -> Result<Option<Timestamp>, D::Error> {
+    struct TimestampVisitor;
+	impl<'d> Visitor<'d> for TimestampVisitor {
+		type Value = Option<Timestamp>;
+
+		fn expecting(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+			write!(fmt, "an f64, u64, or parseable string")
+		}
+
+		fn visit_f64<E: Error>(self, v: f64) -> Result<Option<Timestamp>, E> {
+            Ok(Some(Timestamp {
+                string_repr: v.to_string(),
+                float_repr: v,
+            }))
+        }
+
+		fn visit_str<E: Error>(self, v: &str) -> Result<Option<Timestamp>, E> {
+		    Ok(Some(Timestamp {
+                string_repr: v.to_string(),
```

Comment:
nit: formatting

Reply:
You may want to look at https://github.com/rust-lang-nursery/rustfmt/ to help with your code formatting. Although, I don't think this repo has been through a rustfmt in a while, so it may generate a lot of noise. Probably better to do that in a separate PR.